templates/navbar.njk

```diff
@@ -0,0 +1,135 @@
+{# 
+A macro that adds a menu navigation bar with the provided items.
+It receives:
+- signInHref: the URL of the sign in / register page
+- menuItems: a collection of objects representing individual menu items.
+    Each object must contain properties with the following keys to build an anchor:
+    - id: it's the anchor element id
+    - href: the URL of the page the menu item will point to
+    - displayText: the text to display in the anchor
+- userDisplayText: a text identifying the logged user, e.g. email address or name
+- lang: an optional object with hidden texts and aria labels. If not provided,
+    default texts are used instead. It contains the following keys:
+    - mainNavbar: aria label text for the nav element, default 'The main navigation bar'
+    - clickToExpandOrCollaps: text for the visually hidden part of the anchor text,
+        default 'Click to expand or collapse'
+    - signInOrRegister: the mobile version of sign in anchor, default 'Sign in / Register'
+
+The navbar requires the use of styles provided in the following stylesheet
+(you can either add the provided link to the head section or include specific
+styles in your own stylesheet)
+
+    <link href="{{ cdnHost }}/stylesheets/ch.gov.uk.css" rel="stylesheet"/>
+
+It also requires the following script to be added to the footer
+
+    <script>
+        $(document).ready(function () {
+            $("ul#navigation").addClass('mobile-hidden');
+            $(".js-toggle-nav .js-header-toggle").click(function () {
+                event.preventDefault();
+                $("ul#navigation").toggleClass('mobile-hidden');
+                $("#global-nav").toggleClass('open');
+            });
+        });
+    </script>
+#}
+{% macro addNavbar(signInHref, menuItems, userDisplayText, lang) %}
```

Comment:
you could set `lang` to be by default the empty object:
`lang={}` this will simplify all the `if lang else` to just be an `or`